### PR TITLE
Support lang parameter for Koncorde filters

### PR DESCRIPTION
**Pull request by y-abs**
## What does this PR do ?

Since Kuzzle 2.8.0 supports the "lang" parameter to handle Konkorde filters, we need to add it in SDKs for the API methods who accept it.

### How should this be manually tested?

  - Step 1 :
  - Step 2 :
  - Step 3 :  
  ...

### Boyscout
gitignore node_modules

### CODE REVIEW

File: doc/2/controllers/document/delete-by-query/index.md [hunk 1]
```diff
@@ -49,4 +49,10 @@ Returns a `List<String>` containing the deleted document ids.
 
 ## Usage
 
-<<< ./snippets/delete-by-query.dart
+With the [ElasticSearch Query DSL](https://www.elastic.co/guide/en/elasticsearch/reference/7.3/query-dsl.html) syntax.
```
Comment (Aschen) [inline doc/2/controllers/document/delete-by-query/index.md]:
```suggestion
With the [ElasticSearch Query DSL](https://www.elastic.co/guide/en/elasticsearch/reference/7.4/query-dsl.html) syntax.
```

File: doc/2/controllers/document/search/snippets/search-koncorde.dart [hunk 2]
```diff
@@ -0,0 +1,9 @@
+final result = await kuzzle
+  .document
+  .search('nyc-open-data', 'yellow-taxi', query: {
+    'query': {
```
Comment (Aschen) [inline doc/2/controllers/document/search/snippets/search-koncorde.dart:4]: I'm not sure the top level `query` is needed. I rather think that this should raise an error instead (Kuzzle do raise an error)

File: doc/2/controllers/document/update-by-query/index.md [hunk 3]
```diff
@@ -71,4 +71,10 @@ Each errored document is an object of the `errors` array with the following prop
 
 ## Usage
 
-<<< ./snippets/update-by-query.dart
\ No newline at end of file
+With the [ElasticSearch Query DSL](https://www.elastic.co/guide/en/elasticsearch/reference/7.3/query-dsl.html) syntax.
```
Comment (Aschen) [inline doc/2/controllers/document/update-by-query/index.md]:
```suggestion
With the [ElasticSearch Query DSL](https://www.elastic.co/guide/en/elasticsearch/reference/7.4/query-dsl.html) syntax.
```

File: doc/2/controllers/document/search/index.md [hunk 4]
```diff
@@ -79,4 +79,10 @@ Returns a [SearchResult](/sdk/dart/2/core-classes/search-result) object.
 
 ## Usage
 
-<<< ./snippets/search.dart
+With the [ElasticSearch Query DSL](https://www.elastic.co/guide/en/elasticsearch/reference/7.3/query-dsl.html) syntax.
```
Comment (Aschen) [inline doc/2/controllers/document/search/index.md]:
```suggestion
With the [ElasticSearch Query DSL](https://www.elastic.co/guide/en/elasticsearch/reference/7.4/query-dsl.html) syntax.
```

File: doc/2/controllers/security/search-api-keys/index.md [hunk 5]
```diff
@@ -48,4 +48,10 @@ Returns a [SearchResult](/sdk/dart/2/core-classes/search-result) object.
 
 ## Usage
 
-<<< ./snippets/search-api-keys.dart
+With the [ElasticSearch Query DSL](https://www.elastic.co/guide/en/elasticsearch/reference/7.3/query-dsl.html) syntax.
```
Comment (Aschen) [inline doc/2/controllers/security/search-api-keys/index.md]:
```suggestion
With the [ElasticSearch Query DSL](https://www.elastic.co/guide/en/elasticsearch/reference/7.4/query-dsl.html) syntax.
```

File: doc/2/controllers/security/search-users/index.md [hunk 6]
```diff
@@ -47,4 +47,10 @@ A [`UserSearchResult`](sdk/dart/2/core-classes/search-result) object containing
 
 ## Usage
 
-<<< ./snippets/search-users.dart
+With the [ElasticSearch Query DSL](https://www.elastic.co/guide/en/elasticsearch/reference/7.3/query-dsl.html) syntax.
```
Comment (Aschen) [inline doc/2/controllers/security/search-users/index.md]:
```suggestion
With the [ElasticSearch Query DSL](https://www.elastic.co/guide/en/elasticsearch/reference/7.4/query-dsl.html) syntax.
```

File: doc/2/controllers/security/search-users/snippets/search-users-koncorde.dart [hunk 7]
```diff
@@ -0,0 +1,18 @@
+final result = await kuzzle.security.searchUsers(query: {
+  'query': {
```
Comment (Aschen) [inline doc/2/controllers/security/search-users/snippets/search-users-koncorde.dart:2]: Same here, why we have 2 times the `query` property?

File: doc/2/core-classes/search-result/next/index.md [hunk 8]
```diff
@@ -43,7 +43,13 @@ It can lead to memory leaks if a scroll duration too great is provided, or if to
 You can restrict the scroll session maximum duration under the `services.storage.maxScrollDuration` configuration key.
 :::
 
-<<< ./snippets/scroll.dart
+With the [ElasticSearch Query DSL](https://www.elastic.co/guide/en/elasticsearch/reference/7.3/query-dsl.html) syntax.
```
Comment (Aschen) [inline doc/2/core-classes/search-result/next/index.md]:
```suggestion
With the [ElasticSearch Query DSL](https://www.elastic.co/guide/en/elasticsearch/reference/7.4/query-dsl.html) syntax.
```

File: doc/2/core-classes/search-result/next/index.md [hunk 9]
```diff
@@ -64,4 +70,10 @@ Because this method does not freeze the search results between two calls, there
 It's the fastest pagination method available, but also the less consistent, and it is not possible to retrieve more than 10000 items using it.  
 Above that limit, any call to `next` throws an Exception.
 
-<<< ./snippets/fromsize.dart
+With the [ElasticSearch Query DSL](https://www.elastic.co/guide/en/elasticsearch/reference/7.3/query-dsl.html) syntax.
```
Comment (Aschen) [inline doc/2/core-classes/search-result/next/index.md]:
```suggestion
With the [ElasticSearch Query DSL](https://www.elastic.co/guide/en/elasticsearch/reference/7.4/query-dsl.html) syntax.
```

File: doc/2/core-classes/search-result/next/snippets/fromsize-koncorde.dart [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+}
+
+await kuzzle.document.mCreate('nyc-open-data', 'yellow-taxi', documents,
+  waitForRefresh: true
+);
+
+var res = await kuzzle.document.search(
+  'nyc-open-data',
+  'yellow-taxi',
+  query: { 'query': { 'equals': { 'category': 'suv' } } },
```
Comment (Aschen) [inline doc/2/core-classes/search-result/next/snippets/fromsize-koncorde.dart:14]: Same here, why we have two times `query` ?

File: doc/2/core-classes/search-result/next/snippets/scroll-koncorde.dart [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+}
+
+await kuzzle.document.mCreate('nyc-open-data', 'yellow-taxi', documents,
+  waitForRefresh: true
+);
+
+var res = await kuzzle.document.search(
+  'nyc-open-data',
+  'yellow-taxi',
+  query: { 'query': { 'equals': { 'category': 'suv' } } },
```
Comment (Aschen) [inline doc/2/core-classes/search-result/next/snippets/scroll-koncorde.dart:14]: Ditto

**Review by Aschen — CHANGES_REQUESTED** [review]

File: doc/2/controllers/document/search/snippets/search-koncorde.dart (hunk 2 above)
Comment (y-abs) [reply doc/2/controllers/document/search/snippets/search-koncorde.dart:4]: As we discussed, this is normal here

File: doc/2/controllers/security/search-users/snippets/search-users-koncorde.dart (hunk 7 above)
Comment (y-abs) [reply doc/2/controllers/security/search-users/snippets/search-users-koncorde.dart:2]: As we discussed, this is normal here

File: doc/2/core-classes/search-result/next/snippets/fromsize-koncorde.dart (hunk 10 above)
Comment (y-abs) [reply doc/2/core-classes/search-result/next/snippets/fromsize-koncorde.dart:14]: As we discussed, this is normal here

File: doc/2/core-classes/search-result/next/snippets/scroll-koncorde.dart (hunk 11 above)
Comment (y-abs) [reply doc/2/core-classes/search-result/next/snippets/scroll-koncorde.dart:14]: As we discussed, this is normal here

File: doc/2/controllers/security/search-api-keys/snippets/search-api-keys-koncorde.dart [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+/*
+{
+  "total": 2,
+  "hits": [
+    {
+      "_id": "znEwbG8BJASM_0-bWU-q",
+      "_source": {
+        "description": "LoRa permanent API key",
+        "userId": "john.doe",
+        "expiresAt": -1,
```
Comment (Aschen) [inline doc/2/controllers/security/search-api-keys/snippets/search-api-keys-koncorde.dart:30]: Can you add the missing `fingerprint` property here as well?

**Review by Aschen — CHANGES_REQUESTED** [review]

**Review by xbill82 — APPROVED** [review]

**Review by Aschen — APPROVED** [review]

**Pull request merged**